### PR TITLE
#153 Return Subscription ID as a part of ConfirmResult class

### LINKED ISSUE #153: Return Subscription ID as a part of ConfirmResult class

### PR DESCRIPTION
**Pull request by marclerwick**
Refactored the Get-AzBP methods to return the ConfirmResult type.
Added authentication data, including Subscription ID, to the ConfirmResult type.
Updated Get-AzBP methods to consolidate the Get-* and Get-*Exist methods given the discussions during the design session and updating the methods to return the ConfirmResult type.

Locally, there are problems with the unit tests when creating a new ErrorRecord. All of the research I have done shows that what I wrote is correct, but locally it fails with a constructor error. I'm pushing the code as is, so that the PR unit tests will run, and I can determine if it's a local problem or something more.

Closes #153

### CODE REVIEW

**Comment by marclerwick** [comment]
It looks like the ErrorRecord constructor issue isn't just local. I'll need to find a way to resolve that before this is ready for primetime.

File: BenchPress/Helpers/BenchPress.Azure/AksCluster.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,33 @@
+﻿using module ./public/classes/ConfirmResult.psm1
```
Comment (msha1026) [inline BenchPress/Helpers/BenchPress.Azure/AksCluster.Tests.ps1:1]: Just wondering, was there a specific reason why this file and the test file name was changed?

File: BenchPress/Helpers/BenchPress.Azure/AksCluster.psm1 [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  [OutputType([ConfirmResult])]
+  param (
+    [Parameter(Mandatory=$true)]
+    [string]$AksName,
+
+    [Parameter(Mandatory=$true)]
+    [string]$ResourceGroupName
+  )
+  Begin {
+    Connect-Account
```
Comment (msha1026) [inline BenchPress/Helpers/BenchPress.Azure/AksCluster.psm1:38]: `$ConnectResults = Connect-Account`

File: BenchPress/Helpers/BenchPress.Azure/ContainerRegistry.psm1 [hunk 3]
```diff
@@ -15,64 +17,42 @@ Import-Module $PSScriptRoot/Authentication.psm1
...
     [Parameter(Mandatory=$true)]
     [string]$Name,
 
     [Parameter(Mandatory=$true)]
     [string]$ResourceGroupName
   )
-  $resource = Get-ContainerRegistry -Name $Name -ResourceGroupName $ResourceGroupName
-  return ($null -ne $resource)
+  Begin {
+    Connect-Account
```
Comment (msha1026) [inline BenchPress/Helpers/BenchPress.Azure/ContainerRegistry.psm1:39]: `$ConnectResults = Connect-Account`

File: BenchPress/Helpers/BenchPress.Azure/AppServicePlan.psm1 [hunk 4]
```diff
@@ -15,65 +17,42 @@ Import-Module $PSScriptRoot/Authentication.psm1
...
     [string]$AppServicePlanName,
 
     [Parameter(Mandatory=$true)]
     [string]$ResourceGroupName
   )
-
-  $resource = Get-AppServicePlan -AppServicePlanName $AppServicePlanName -ResourceGroupName $ResourceGroupName
-  return ($null -ne $resource)
+  Begin {
+    Connect-Account
```
Comment (msha1026) [inline BenchPress/Helpers/BenchPress.Azure/AppServicePlan.psm1:39]: `$ConnectResults = Connect-Account`

**Review by msha1026 — APPROVED** [review]
Other than the few small changes that need to be made, this looks great! Approving :)

File: BenchPress/Helpers/BenchPress.Azure/AksCluster.Tests.ps1 (hunk 1 above)
Comment (marclerwick) [reply BenchPress/Helpers/BenchPress.Azure/AksCluster.Tests.ps1:1]: In my attempt to follow best practices for project folder structure, when I created my two new classes I put them in the `./public/classes` folder. I decided to go ahead and move this one at the same time.

**Comment by marclerwick** [comment]
> It looks like the ErrorRecord constructor issue isn't just local. I'll need to find a way to resolve that before this is ready for primetime.

This has been resolved. Thanks @msha1026 for the pairing!

File: BenchPress/Helpers/BenchPress.Azure/AksCluster.Tests.ps1 (hunk 1 above)
Comment (msha1026) [reply BenchPress/Helpers/BenchPress.Azure/AksCluster.Tests.ps1:1]: Sorry, don't know to comment on overall files in Github like I can with AzDO. I should be more specific. Was wondering why the `AKSCluster.ps1` and `AKSCluster.Tests.ps1` files were renamed to `AksCluster`? I do think it fits CamelCase better.

Comment (marclerwick) [reply BenchPress/Helpers/BenchPress.Azure/AksCluster.Tests.ps1:1]: Ah, sorry, completely missed the point of the question... yes, it was for best practices for camel casing. Plus, it reflects Microsoft's Az PowerShell cmdlet naming.

There are typically two schools of thought when it comes to Pascal and camel casing, and neither of them supports three capital letters in a row. One school of thought is that any acronym, regardless of length, should only ever have the initial character capitalized. The other school of thought is that acronyms that are two characters long can be capitalized.

Microsoft is in the latter camp: https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions#capitalization-rules-for-identifiers

**Review by ozboms — APPROVED** [review]

**Pull request merged**